system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -34,45 +33,48 @@ export class TopSiteForm extends React.PureComponent {
     this.props.onClose();
   }
 
-  onAddButtonClick(ev) {
+  onDoneButtonClick(ev) {
     ev.preventDefault();
+
     if (this.validateForm()) {
-      let site = {url: this.cleanUrl()};
+      const site = {url: this.cleanUrl()};
       if (this.state.label !== "") {
         site.label = this.state.label;
       }
-      this.props.dispatch(ac.SendToMain({
-        type: at.TOP_SITES_INSERT,
-        data: {site}
-      }));
-      this.props.dispatch(ac.UserEvent({
-        source: TOP_SITES_SOURCE,
-        event: "TOP_SITES_ADD"
-      }));
-      this.props.onClose();
-    }
-  }
 
-  onSaveButtonClick(ev) {
-    ev.preventDefault();
-    if (this.validateForm()) {
-      let site = {url: this.cleanUrl()};
-      if (this.state.label !== "") {
-        site.label = this.state.label;
+      if (this.props.index >= 0) {
```

Comment:
Similarly we want a comment here as why we're using a check different from the UI bits.